backslash/contrib/slash_plugin.py

```diff
@@ -145,9 +145,35 @@ def session_interrupt(self):
         if self.session is not None:
             self.session.report_interrupted()
 
+    @slash.plugins.register_if(_HAS_REPORT_UPCOMING_TESTS)
+    @handle_exceptions
+    def report_upcoming_tests(self):
+        tests = slash.context.session.get_tests()
+        tests_metadata = []
+        for test in tests:
+            test_info = self._get_test_info(test)
+            current = {'test_logical_id':test.__slash__.id,
+                       'file_name':test_info['file_name'],
+                       'name':test_info['name'],
+                       'class_name':test_info['class_name']
+                      }
+            if 'variation' in test_info:
+                current['variation'] = test_info['variation']
+            tests_metadata.append(current)
+
+        tests_count = 0
+        batch_size = 100
+        try:
+            while tests_count < len(tests_metadata):
+                self.session.report_upcoming_tests(tests_metadata[tests_count:tests_count+batch_size])
+                tests_count += batch_size
+        except requests.exceptions.HTTPError as e:
```

Comment:
What happens if we get an internal error (e.g. 500) here? will report_test_start still work?